src/sage/geometry/polyhedron/combinatorial_polyhedron/base.pyx

```diff
@@ -1469,23 +1458,21 @@ cdef class CombinatorialPolyhedron(SageObject):
             def f(size_t i):
                 return smallInteger(i)
 
-        # Getting the indices of the `i`-th ridge.
-        def facet_one(size_t i):
-            return f(self._get_edge(self._ridges, i, 0))
-
-        def facet_two(size_t i):
-            return f(self._get_edge(self._ridges, i, 1))
+        cdef pair_s ridge
 
-        cdef size_t j
         if add_equations and names:
-            # Also getting the equations for each facet.
-            return tuple(
-                (((facet_one(i),) + self.equations()),
-                 ((facet_two(i),) + self.equations()))
-                for i in range(n_ridges))
+            def get_ridge(size_t i):
+                ridge = self._ridges.get(i)[0]
+                return ((f(ridge.first),) + self.equations(),
+                        (f(ridge.second),) + self.equations())
+
         else:
-            return tuple((facet_one(i), facet_two(i))
-                         for i in range(n_ridges))
+            def get_ridge(size_t i):
+                ridge = self._ridges.get(i)[0]
+                return (f(ridge.first), f(ridge.second))
+
+        cdef size_t j
+        return tuple(get_ridge(j) for j in range(n_ridges))
```

Comment:
I am not sure about these changes. They seem to slow it down (extra Python function calls) and is at best a net zero improvement to readability (I would actually argue it makes it more complicated). Although I guess you are halving the number of function calls, but I think it would be better to just have the two cases do a `return`.

Reply:
Improved in 2367bc9685579e0a554a1371f6b919e532ab6845.